Cython/Compiler/ExprNodes.py

```diff
@@ -8374,16 +8387,26 @@ def generate_special_parallel_unpacking_code(self, code, rhs, use_loop):
         if len(sequence_types) == 2:
             code.putln("if (likely(Py%s_CheckExact(sequence))) {" % sequence_types[0])
         for i, item in enumerate(self.unpacked_items):
-            code.putln("%s = Py%s_GET_ITEM(sequence, %d); " % (
-                item.result(), sequence_types[0], i))
+            if sequence_types[0] == "List":
+                code.putln("%s = __Pyx_PyList_GetItemRef(sequence, %d); " % (
+                    item.result(), i))
+                code.put_xgotref(item.result(), item.ctype())
+            else:  # Tuple
+                code.putln("%s = PyTuple_GET_ITEM(sequence, %d); " % (
+                    item.result(), i))
+                code.put_incref(item.result(), item.ctype())
         if len(sequence_types) == 2:
             code.putln("} else {")
             for i, item in enumerate(self.unpacked_items):
-                code.putln("%s = Py%s_GET_ITEM(sequence, %d); " % (
-                    item.result(), sequence_types[1], i))
+                if sequence_types[1] == "List":
+                    code.putln("%s = __Pyx_PyList_GetItemRef(sequence, %d); " % (
+                        item.result(), i))
+                    code.put_xgotref(item.result(), item.ctype())
+                else:  # Tuple
+                    code.putln("%s = PyTuple_GET_ITEM(sequence, %d); " % (
+                        item.result(), i))
+                    code.put_incref(item.result(), item.ctype())
```

Comment:
I think the space in the end of the (original) template string probably intended (or was left over from) allowing to merge the lines, so it's either `code.put("… ")` with trailing space or `code.putln("…")` without it. I don't mind either way but would probably lean towards using separate lines since that keeps the difference between gotref and incref visible. I.e. remove the trailing space here and above.